src/app/firedancer-dev/commands/catchup.c

```diff
@@ -0,0 +1,199 @@
+#include "../../shared/fd_config.h"
+#include "../../shared/fd_action.h"
+
+#include "../../../disco/topo/fd_topo.h"
+#include "../../../discof/repair/fd_repair_tile.c"
+#include "../../../waltz/resolv/fd_io_readline.h"
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+struct fd_location_info {
+  ulong ip4_addr;         /* for map key convenience */
+  char location[ 128 ];
+};
+typedef struct fd_location_info fd_location_info_t;
+
+#define MAP_NAME    fd_location_table
+#define MAP_T       fd_location_info_t
+#define MAP_KEY     ip4_addr
+#define MAP_MEMOIZE 0
+#include "../../../util/tmpl/fd_map_dynamic.c"
+
+fd_location_info_t * location_table;
+fd_peer_t peers_copy[ FD_ACTIVE_KEY_MAX ];
+
+static void
+catchup_cmd_args( int *    pargc,
+                  char *** pargv,
+                  args_t * args ) {
+  if( FD_UNLIKELY( !*pargc ) ) FD_LOG_ERR(( "usage: catchup [--iptable-path <iptable_path>]" ));
+  fd_cstr_fini( fd_cstr_append_cstr_safe( fd_cstr_init( args->catchup.iptable_path ), fd_env_strip_cmdline_cstr( pargc, pargv, "--iptable-path", NULL, "unknown" ), sizeof(args->catchup.iptable_path)-1UL ) );
+  FD_LOG_NOTICE(( "catchup iptable_path %s", args->catchup.iptable_path ));
+}
+
+static void
+sort_peers_by_latency( fd_active_elem_t * active_table, fd_peer_t * peer_arr, ulong peer_cnt ) {
+  for( uint i = 0; i < peer_cnt - 1; i++ ) {
+    int swapped = 0;
+    for( uint j = 0; j < peer_cnt - 1 - i; j++ ) {
+      fd_active_elem_t const * active_j  = fd_active_table_query_const( active_table, &peer_arr[ j ].key,     NULL );
+      fd_active_elem_t const * active_j1 = fd_active_table_query_const( active_table, &peer_arr[ j + 1 ].key, NULL );
+
+      /* Skip peers with no responses */
+      double latency_j  = 10e9;
+      double latency_j1 = 10e9;
+      if( FD_LIKELY( active_j && active_j->resp_cnt > 0   ) ) latency_j  = ((double)active_j->total_latency / (double)active_j->resp_cnt);
+      if( FD_LIKELY( active_j1 && active_j1->resp_cnt > 0 ) ) latency_j1 = ((double)active_j1->total_latency / (double)active_j1->resp_cnt);
+
+      /* Swap if j has higher latency than j+1 */
+      if( latency_j > latency_j1 ) {
+        fd_peer_t temp    = peer_arr[ j ];
+        peer_arr[ j ]     = peer_arr[ j + 1 ];
+        peer_arr[ j + 1 ] = temp;
+        swapped           = 1;
+      }
+    }
+    if( !swapped ) break;
+  }
+}
+
+#define MAX_WIDTH 120
+static char dashes[MAX_WIDTH + 1] = "========================================================================================================================";
+static char spaces[MAX_WIDTH + 1] = "                                                                                                                        ";
+
+static void
+print_catchup_slots( fd_wksp_t * repair_tile_wksp, fd_repair_tile_ctx_t * repair_ctx ) {
```

Comment:
so i was actually thinking this would be its own module API (fd_catchup or something like that), and the data structure would be encapsulated within that

it would like in repair tile ctx's mem -- but i like the idea of a subcommand to query it